.github/workflows/ci.yml

```diff
@@ -12,14 +12,23 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        java: [21]
+        java: [8, 11, 17, 21]
     steps:
     - uses: actions/checkout@v2
     - name: Setup java
-      uses: actions/setup-java@v1
+      uses: actions/setup-java@v4
       with:
-        java-version: ${{ matrix.java }}
+        distribution: temurin
+        java-version: |
+          ${{ matrix.java }}
+          21
```

Comment:
Is this intentionally left or just forgot to remove line?

Reply:
Intentionally. We need to install 21 always to build the project, regardless test-jdk version.
Last one will be used as the global java version (refs: https://github.com/actions/setup-java?tab=readme-ov-file#install-multiple-jdks)

Reply:
ah-ok. now understand how it works